src/query/storages/parquet/src/parquet_rs/parquet_table/partition.rs

```diff
@@ -238,7 +239,11 @@ impl ParquetRSTable {
             let copy_status = copy_status.clone();
             let leaf_fields = self.leaf_fields.clone();
             let topk = topk.clone();
-            let is_copy = matches!(ctx.get_query_kind(), QueryKind::CopyIntoTable);
+            let is_remote_query =
```

Comment:
This does not work: COPY/REPLACE etc. can also read the parquet file from the stage, it's hard to check a query is read only. 
So we should lower this PR's priority if a better method is found.
Thanks.